arch/mips/include/kernel_arch_func.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2017 Imagination Technologies Ltd.
+ */
+
+/**
+ * @file
+ * @brief Private kernel definitions
+ *
+ * This file contains private kernel function/macro definitions and various
+ * other definitions for the MIPS processor architecture.
+ */
+
+#ifndef _kernel_arch_func__h_
+#define _kernel_arch_func__h_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <soc.h>
+
+#ifndef _ASMLANGUAGE
+void k_cpu_idle(void);
+void k_cpu_atomic_idle(unsigned int key);
+
+static ALWAYS_INLINE void
+_arch_switch_to_main_thread(struct k_thread *main_thread, char *main_stack,
+			    size_t main_stack_size, k_thread_entry_t _main)
+{
+	/* the aligned attribute here is the only difference between
+	 * this and the standard switch
+	 */
+	struct k_thread dummy_thread_memory __aligned(8);
```

Comment:
what's special about the dummy thread that it has to be aligned(8), unlike all other struct k_thread in the system?